File: 1615-range-sum-of-sorted-subarray-sums/1615-range-sum-of-sorted-subarray-sums.py
class Solution:
    def rangeSum(self, nums: List[int], n: int, left: int, right: int) -> int:
        m=1000000007
        allSubSum = []

        for i in range(0,n):
            sum = 0
            for j in range(i,n):
                sum = (sum + nums[j])
                allSubSum.append(sum)

        allSubSum.sort()

        newSum = 0
        for i in range(left-1, right):
            newSum = (newSum + allSubSum[i])

        return (newSum)%m

File: 1615-range-sum-of-sorted-subarray-sums/1615-range-sum-of-sorted-subarray-sums.cpp
class Solution {
public:
    int rangeSum(vector<int>& nums, int n, int left, int right) {
        int m = 1000000000+7;
        vector<int>allSubSum;

        for(int i=0; i<n; i++)
        {
            int sum = 0;
            for(int j=i; j<n; j++)
            {
                sum = (sum + nums[j])%m;
                allSubSum.push_back(sum);
            }
        }
        sort(allSubSum.begin(), allSubSum.end());

        int newSum=0;
        for(int i=left-1; i<=right-1; i++)
        {
            newSum = (newSum+allSubSum[i])%m;
        }
        return newSum;
    }
};